File: poc-challenges/250730/src/test/java/com/tdd/domain/FactorialTest.java
package com.tdd.domain;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public class FactorialTest {
	@Test
	void factorial_test() {
		assertEquals(1, Factorial.fac(1));
		assertEquals(2, Factorial.fac(2));
	}

	private static class Factorial {
		public static int fac(int i) {
			return i;
		}
	}
}

File: poc-challenges/250917/src/test/kotlin/com/currenjin/tdd/FactorialTest.kt
package com.currenjin.tdd

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class FactorialTest {
    @ParameterizedTest
    @CsvSource("1, 1", "2, 2", "6, 3", "24, 4")
    fun factorial(expected: Int, input: Int) {
        assertEquals(expected, Factorial.fac(input))
    }
}